### DISCUSSION TITLE
v9.5  (2022/11)

### DISCUSSION THREAD

**Discussion by Mizux** [google/or-tools#3569] [Announces]
# Platforms
* Add Debian-SID support (#3457)
* Add Ubuntu 22.10
* Drop Python 3.6 macOS x86_64 (Intel) 
* Add Python 3.11 wheel (#3515)

# Linear Solver
* add `knapsack_interface.cc`.
* Move model_builder API under the linear_solver directory (headers and samples).
* Add support for Gurobi 10

# Routing
* Liberate few parser for various routing challenge.

# CP-SAT
* Presolve improvements: max(array), Boolean constraints, linear constraints.
* Interleaved search should be deterministic in parallel.
* Linear cuts: cleanup square and int_prod cuts; rewrite cut pipeline.
* Fingerprint input model and solution (visible in the log).
* Scheduling improvements
* The usual bunch of bugfixes (crash during presolve, crash in cuts, infeasible solutions, model infeasible in LNS).

# GLOP
* Speedup by rewriting the linear algebra, as well as the pivot selection rule

<hr /><em>This discussion was created from the release <a href='https://github.com/google/or-tools/releases/tag/v9.5'>v9.5  (2022/11)</a>.</em>

**Comment by CADBIMDeveloper** [comment]
Hi @Mizux , when do you plan to publish nuget packages?

**Reply by Mizux** [reply]
Tomorrow morning, at home today so I can't upload them (forget my yubikey at office `-_-`).
BTW you can find them on the release page, dl them in a directory and tweak you .csproj to find them...
```csproj
<PropertyGroup> 
 <RestoreSources>local_path_to_downloaded_packages_directory;$(RestoreSources);https://api.nuget.org/v3/index.json</RestoreSources>
</PropertyGroup>
```
EDIT: somehow manage to upload them, feedback welcome !

**Reply by CADBIMDeveloper** [reply]
Thanks!
